Don't impose too recent cmake version: only enable optional feature if possible.
While finding the most suited python interpreter is nice, it will still work
without this, especially inside robotpkg.

--- rosidl_generator_type_description/cmake/rosidl_generator_type_description_generate_interfaces.cmake~	2024-10-03 16:58:05.000000000 +0200
+++ rosidl_generator_type_description/cmake/rosidl_generator_type_description_generate_interfaces.cmake	2025-04-29 23:25:02.495359211 +0200
@@ -24,9 +24,10 @@
 # the search will prefer /usr/bin/python3 over /usr/bin/python3.11.  And that
 # latter functionality is only available in CMake 3.20 or later, so we need
 # at least that version.
-cmake_minimum_required(VERSION 3.20)
-cmake_policy(SET CMP0094 NEW)
-set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.20")
+  cmake_policy(SET CMP0094 NEW)
+  set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+endif()
 
 find_package(Python3 REQUIRED COMPONENTS Interpreter)
 
--- rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake~	2024-10-03 16:58:05.000000000 +0200
+++ rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake	2025-04-29 23:39:13.829324811 +0200
@@ -104,9 +104,10 @@
 # the search will prefer /usr/bin/python3 over /usr/bin/python3.11.  And that
 # latter functionality is only available in CMake 3.20 or later, so we need
 # at least that version.
-cmake_minimum_required(VERSION 3.20)
-cmake_policy(SET CMP0094 NEW)
-set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.20")
+  cmake_policy(SET CMP0094 NEW)
+  set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+endif()
 
 find_package(Python3 REQUIRED COMPONENTS Interpreter)
 
--- rosidl_generator_cpp/cmake/rosidl_generator_cpp_generate_interfaces.cmake~	2024-10-03 16:58:05.000000000 +0200
+++ rosidl_generator_cpp/cmake/rosidl_generator_cpp_generate_interfaces.cmake	2025-04-29 23:40:10.547318378 +0200
@@ -106,9 +106,10 @@
 # the search will prefer /usr/bin/python3 over /usr/bin/python3.11.  And that
 # latter functionality is only available in CMake 3.20 or later, so we need
 # at least that version.
-cmake_minimum_required(VERSION 3.20)
-cmake_policy(SET CMP0094 NEW)
-set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.20")
+  cmake_policy(SET CMP0094 NEW)
+  set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+endif()
 
 find_package(Python3 REQUIRED COMPONENTS Interpreter)
 
--- rosidl_typesupport_introspection_c/cmake/rosidl_typesupport_introspection_c_generate_interfaces.cmake~	2024-10-03 16:58:05.000000000 +0200
+++ rosidl_typesupport_introspection_c/cmake/rosidl_typesupport_introspection_c_generate_interfaces.cmake	2025-04-29 23:40:51.414937138 +0200
@@ -88,9 +88,10 @@
 # the search will prefer /usr/bin/python3 over /usr/bin/python3.11.  And that
 # latter functionality is only available in CMake 3.20 or later, so we need
 # at least that version.
-cmake_minimum_required(VERSION 3.20)
-cmake_policy(SET CMP0094 NEW)
-set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.20")
+  cmake_policy(SET CMP0094 NEW)
+  set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+endif()
 
 find_package(Python3 REQUIRED COMPONENTS Interpreter)
 
--- rosidl_typesupport_introspection_cpp/cmake/rosidl_typesupport_introspection_cpp_generate_interfaces.cmake~	2024-10-03 16:58:05.000000000 +0200
+++ rosidl_typesupport_introspection_cpp/cmake/rosidl_typesupport_introspection_cpp_generate_interfaces.cmake	2025-04-29 23:41:29.893648207 +0200
@@ -88,9 +88,10 @@
 # the search will prefer /usr/bin/python3 over /usr/bin/python3.11.  And that
 # latter functionality is only available in CMake 3.20 or later, so we need
 # at least that version.
-cmake_minimum_required(VERSION 3.20)
-cmake_policy(SET CMP0094 NEW)
-set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.20")
+  cmake_policy(SET CMP0094 NEW)
+  set(Python3_FIND_UNVERSIONED_NAMES FIRST)
+endif()
 
 find_package(Python3 REQUIRED COMPONENTS Interpreter)
 
